src/main/java/nl/utwente/presto/tezos/TezosMetadata.java

```diff
@@ -99,13 +99,16 @@ public TezosMetadata(
                                         : findBlockByTimestamp((Long) low.getValue(), -1L);
                                 long endBlock = high.isUpperUnbounded() ? -1L
                                         : findBlockByTimestamp((Long) high.getValue(), 1L);
-                                builder.add(new TezosBlockRange(startBlock, endBlock));
+                                builder.add(new TezosRange(startBlock, endBlock));
                             } catch (IOException e) {
                                 throw new IllegalStateException("Unable to find block by timestamp");
                             }
                         });
                         log.info(entry.getValue().getValues().toString(null));
                         break;
+
+                    case "proposal_id":
```

Comment:
There's an empty case here. This should probably create a TezosRange based on the proposal_id